bindings/src/main/java/io/pravega/storage/azure/AzureBlobClientImpl.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.storage.azure;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.AppendBlobItem;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.AppendBlobRequestConditions;
+import com.azure.storage.blob.specialized.AppendBlobClient;
+import com.azure.storage.blob.specialized.BlobClientBase;
+import com.google.common.base.Preconditions;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * {@link AzureBlobClientImpl} class provides implementation of {@link AzureClient} methods.
+ */
+@Slf4j
+public class AzureBlobClientImpl implements AzureClient {
+
+    /**
+     * Client to a container which contains operations on the blob container.
+     */
+    private final BlobContainerClient blobContainerClient;
+
+    /**
+     * Providing different configurations for Azure Storage component.
+     */
+    private final AzureStorageConfig config;
+
+    /**
+     * Automatically updating the value to close the resources.
+     */
+    private final AtomicBoolean closed = new AtomicBoolean();
+
+    /**
+     * The class provides implementation of the methods declared in AzureClient.
+     * AzureBlobClientImpl constructor for initializing the following parameters and pre-checking the conditions.
+     * @param config Configuration for Azure Storage Account.
+     * @param blobContainerClient client to a container performing operations on container.
+     */
+    public AzureBlobClientImpl(AzureStorageConfig config, BlobContainerClient blobContainerClient) {
+        this.config = Preconditions.checkNotNull(config, "config");
+        Preconditions.checkNotNull(config, "");
```

Comment:
These seems to be duplicate.

Reply:
Sure, Amit. Thanks for bringing it to my notice.